### ISSUE TITLE
Ability to use aliases in user-defined functions (CREATE FUNCTION statement)

### ISSUE THREAD

**Issue by vpanfilov** [ClickHouse/ClickHouse#29188]
Starting with Clickhouse  version 21.10, it is possible to define custom function with lambda-expression inside CREATE FUNCTION statement. But it is not possible to use aliases inside lambda-expressions:
```sql
CREATE FUNCTION testAlias AS x -> (((x * 2) AS x_doubled) + x_doubled)

Received exception from server (version 21.10.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Identifier `x_doubled` does not exist in arguments. (UNKNOWN_IDENTIFIER)
```

It would be perfect if it was possible to place all aliases before main lambda expression. I mean something like that:
```sql
CREATE FUNCTION testAlias AS x ->
WITH
  x * 2 AS x_doubled,
  pow(x,2) AS x_squared
RETURN
  x_doubled + x_squared
```

**Comment by kylhuk** [comment]
Agree, that would be useful. Additionally, if you use a UDF, the column header does not use the function name and aliases on function names do not work, e.g. `SELECT my_function(column1) AS custom_header`

So if no alias is given, the column header should be `my_function(column1)`, if an alias is given, the header should use the alias (e.g. `custom_header`)

**Comment by kitaisreal** [comment]
Need to wait until https://github.com/ClickHouse/ClickHouse/issues/23194 will be implemented.
